src/backend/storage/smgr/smgr.c

```diff
@@ -754,3 +780,16 @@ AtEOXact_SMgr(void)
 		smgrclose(rel);
 	}
 }
+
+SMgrImpl
+add_smgr_kind(void)
```

Comment:
it better to add some comments for it. 
To use custom smgr,  it should first call add_smgr_kind() to get a slot, and then initialize the slot.

Reply:
OK